### DISCUSSION TITLE
A note on Gyromina + slash commands

### DISCUSSION THREAD

**Discussion by Lowie375** [Lowie375/Gyromina#84] [Announcements]
### Since Gyromina v1.2 has recently entered the beta phase of development, I figured I'd give an update on my stance regarding slash commands.

As of now (on [`indev-branch`](/Lowie375/Gyromina/tree/indev-branch)), 5 existing commands have been made slash-compatible (`/help`, `/info`, `/ping`, `/pong`, and `/uptime`), and 1 new slash command (`/getprefix`) has been added. These commands will be pushed with the release of Gyromina v1.2.0.

My current plan is to gradually migrate all other existing commands to a slash-compatible framework, bundling them in with either patch releases (i.e. `v1.2.x`) or minor releases (i.e. `v1.x.x`). I've decided to take this approach due to the complexity of some existing commands (for instance: `play`, a command handler within a command handler) and the challenges that would come with attempting to migrate them.
More simply put, I'd rather push an otherwise feature-complete update with partial slash command support than delay it more than necessary to migrate *everything* at once.

It's also worth noting that Discord has currently placed a 100 slash command cap per bot application. As this point in time, Gyromina is at about one-fifth of that number, so I don't expect this to impede development for the foreseeable future. If it does, however, I may have to begin strategically determining which commands are the most important to have as slash commands.

**TL;DR: Slash commands are hard. Partial support will come in v1.2, full support *will* come sometime in the near future.**

**Comment by Lowie375** [comment]
### A brief (post-v1.2) update
Currently, Discord is planning a full bot structure migration: from message-based to interaction-based. Because of this, Gyromina will *definitely* be getting slash command support, but again, gradually.

My current plans for v1.3.0 consist of migrating more commands to slash-compatible frameworks and working on some more backend changes, so a wave of slash commands will likely come with that release.
